plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/command/CommandConfiguration.java

```diff
@@ -55,4 +55,50 @@ public CommandType getType() {
     /** Returns command line to execute in machine. */
     @NotNull
     public abstract String toCommandLine();
+    
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || !(o instanceof CommandConfiguration)) {
+                return false;
+            }
+            CommandConfiguration other = (CommandConfiguration)o;
+            if (this.getName() != null) {
+                if (!this.getName().equals(other.getName())) {
+                    return false;
+                }
+            } else {
+                if (other.getName() != null) {
+                    return false;
+                }
+            }
+            if (this.getType() != null) {
+                if (!this.getType().getId().equals(other.getType().getId())) {
+                    return false;
+                }
+            } else {
+                if (other.getType() != null) {
+                    return false;
+                }
+            }
+            if (this.toCommandLine() != null) {
+                if (!this.toCommandLine().equals(other.toCommandLine())) {
+                    return false;
+                }
+            } else {
+                if (other.toCommandLine() != null) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        @Override
+        public int hashCode() {
+            int hash = 7;
+            hash = hash * 31 + (name != null ? name.hashCode() : 0);
```

Comment:
Use *Objects.hashCode(name)* instead of *(name != null ? name.hashCode() : 0)*
And the same for other similar code